### ISSUE TITLE
Check that indices can be specified as expected

### ISSUE THREAD

**Issue by grst** [nf-core/scrnaseq#393]
### Description of the bug

Specifying indices should work as expected from a nf-core pipeline, i.e.
 * igenomes should be respected
 * it shall be possible to specify pre-computed indices and they shall be actually used
 * it shall be possible to generate indices from fastq/gtf otherwise. 

There are a couple of issues related to that and it's probably best to address them all in one go and add tests where it makes sense. 

Related issues:
https://github.com/nf-core/scrnaseq/pull/390
https://github.com/nf-core/scrnaseq/issues/339
https://github.com/nf-core/scrnaseq/issues/345

**Comment by kopichris** [comment]
Happy to look into this if the issue is still open 👍

**Comment by grst** [comment]
It would be great if you could take a look, but it's better to wait until #369 is merged because @fmalmeida already made some changes to this (but without the aim to address this entirely). 

If you already want to get startet, you could also try out the branch from #369 - It would anyway be good to get some additional feedback on this because it entails quite some changes.

**Comment by kopichris** [comment]
Happy to play around with the #369 branch until it gets merged, thanks for the heads up!

**Comment by kopichris** [comment]
That's great news!  I'll test the behavior of #369 and report back by Friday!

**Comment by kopichris** [comment]
Hi @grst, I reviewed and tested the following issues from the `dev` branch:

[#330]: I ran the `test,conda` profile and received the following error in the `NFCORE_SCRNASEQ:SCRNASEQ:H5AD_CONVERSION:
ANNDATAR_CONVERT` process: `Error in library(anndataR) : there is no package called ‘anndataR’`. I suspect this is because there is no `conda` recipe in the [ANNDATAR_CONVERT](https://github.com/nf-core/scrnaseq/blob/dev/modules/local/anndatar_convert.nf) process.

[#339 and #345]: Pipeline completed successfully when specifying the `--fasta` and `--gtf` parameters.

[#366]: Pipeline completed successfully when using the `test_full,docker` profile with the `--skip_emptydrops` parameter.  Pipeline failed on time when running with `emptydrops`.  This was likely because the `emptydrops` workflow is optimized for [GPUs](https://cellbender.readthedocs.io/en/latest/tutorial/index.html#run-remove-background), which I don't have on my system.  It may be worth skipping the `emptydrops` workflow by default in the `test_full` profile.

Anything else you think we should look at?

**Comment by grst** [comment]
> I ran the test,conda profile and received the following error in the NFCORE_SCRNASEQ:SCRNASEQ:H5AD_CONVERSION: ANNDATAR_CONVERT process: Error in library(anndataR) : there is no package called ‘anndataR’. I suspect this is because there is no conda recipe in the [ANNDATAR_CONVERT](https://github.com/nf-core/scrnaseq/blob/dev/modules/local/anndatar_convert.nf) process.

Yes, this is expected as anndataR uses a custom container for now. So running with the `conda` profile is currently not supported (same as for cellranger). 

> [https://github.com/nf-core/scrnaseq/issues/366]: Pipeline completed successfully when using the test_full,docker profile with the --skip_emptydrops parameter. Pipeline failed on time when running with emptydrops. This was likely because the emptydrops workflow is optimized for [GPUs](https://cellbender.readthedocs.io/en/latest/tutorial/index.html#run-remove-background), which I don't have on my system. It may be worth skipping the emptydrops workflow by default in the test_full profile.

That's a good point! Cellbender should still finish on CPU if one throws enough cores and time at it, but I think you are right and we should disable it at least for the test profile.

**Comment by nictru** [comment]
I think this has been solved for a long time, and the discussion has diverged to different topics. Closing this, as I recently noticed this thread when working at #545
